src/pymorize/units.py

```diff
@@ -9,189 +9,246 @@
 provide a mapping of what this dimensionless quantity represents and that
 is used for the conversion. `data/dimensionless_mappings.yaml` contains some
 examples on how the mapping is written.
+
+`handle_unit_conversion` is the only function users care about as it handles
+the unit conversion of a DataArray according to a Rule. The rest of the functions
+in this module are support functions.
 """
 
 import re
 from typing import Pattern, Union
 
 import cf_xarray.units  # noqa: F401 # pylint: disable=unused-import
+import pint
 import pint_xarray
-import xarray as xr
 from chemicals import periodic_table
 
 from .logging import logger
-from .rule import Rule
 
 ureg = pint_xarray.unit_registry
 
 
+def _get_units(da, rule):
+    """
+    Get the units from a DataArray and a Rule.
+
+    This function extracts the units from a DataArray and a Rule. If the Rule
+    contains a model_units entry, this takes precedence over the units defined
+    in the dataset. The function also handles dimensionless units by looking up
+    a unit alias in the dimensionless_unit_mappings dictionary of the Rule.
+
+    Parameters
+    ----------
+    da : xarray.DataArray
+        The DataArray to extract the units from.
+    rule : dict
+        The Rule to extract the units from.
+
+    Returns
+    -------
+    from_unit : str
+        The unit of the DataArray.
+    to_unit : str
+        The unit to convert the DataArray to.
+    to_unit_alias : str
+        The unit alias used for representing the to_unit.
+    """
+    model_unit = rule.get("model_unit", None)
+    from_unit = da.attrs.get("units", None)
+    if model_unit is not None:
+        print(
```

Comment:
Please use `logger` instead.
```suggestion
        logger.info(
```